pkg/oauthserver/oauthserver/auth.go

```diff
@@ -140,7 +140,7 @@ func (c *OAuthServerConfig) WithOAuth(handler http.Handler) (http.Handler, error
 	)
 	server.Install(mux, urls.OpenShiftOAuthAPIPrefix)
 
-	tokenRequestEndpoints := tokenrequest.NewEndpoints(c.ExtraOAuthConfig.Options.MasterPublicURL, c.getOsinOAuthClient)
+	tokenRequestEndpoints := tokenrequest.NewTokenRequest(c.ExtraOAuthConfig.Options.MasterPublicURL, c.getOsinOAuthClient, c.ExtraOAuthConfig.OAuthAccessTokenClient)
```

Comment:
ffb4fa19c2f81e85e86f866603da103b69913e88 should not have `c.ExtraOAuthConfig.OAuthAccessTokenClient`